process/block/baseProcess.go

```diff
@@ -2075,3 +2104,64 @@ func (bp *baseProcessor) setNonceOfFirstCommittedBlock(nonce uint64) {
 	bp.nonceOfFirstCommittedBlock.HasValue = true
 	bp.nonceOfFirstCommittedBlock.Value = nonce
 }
+
+func (bp *baseProcessor) handleBlockProcessingCutoff(header data.HeaderHandler) error {
+	if !bp.blockProcessingCutoffConfig.Enabled || check.IfNil(header) {
+		return nil
+	}
+
+	cutOffFunction := getCutoffFunction(bp.blockProcessingCutoffConfig)
+	value := bp.blockProcessingCutoffConfig.Value
+
+	switch common.BlockProcessingCutoffTrigger(bp.blockProcessingCutoffConfig.CutoffTrigger) {
+	case common.BlockProcessingCutoffByRound:
+		if header.GetRound() >= value {
+			err := cutOffFunction("round", header.GetRound())
+			if err != nil {
+				return err
+			}
+		}
+	case common.BlockProcessingCutoffByNonce:
+		if header.GetNonce() >= value {
+			err := cutOffFunction("nonce", header.GetNonce())
+			if err != nil {
+				return err
+			}
+		}
+	case common.BlockProcessingCutoffByEpoch:
+		if header.GetEpoch() >= uint32(value) {
+			err := cutOffFunction("epoch", header.GetEpoch())
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func getCutoffFunction(cfg config.BlockProcessingCutoffConfig) func(printArgs ...interface{}) error {
+	processErr := fmt.Errorf("block processing cuttoff - error")
+	if cfg.Mode == common.BlockProcessingCutoffModeProcessError {
+		return func(printArgs ...interface{}) error {
```

Comment:
when refactoring and adding the new component, instead of having this anonymous functions, we can have proper functions.

Reply:
for reusing code, I've kept this section

Reply:
Please refactor, it can be done without messing with function pointers